### PR TITLE
[4.0] Remove jQuery UI from J4

### PR DESCRIPTION
**Pull request by wilsonge**
Related issue https://github.com/joomla/joomla-cms/issues/18794

### Summary of Changes
There hasn't been a release of jQuery UI since 2016. It's also unused in the J4 codebase. As a result this removes it from J4 as it's clearly not going to be supported going forward (even though as far as I know there's no official announcement).

This comes with a deprecation PR for 3.10 https://github.com/joomla/joomla-cms/pull/28342

### Testing Instructions
Code Review

### Documentation Changes Required
Document Removal

### CODE REVIEW

**Comment by brianteeman** [comment]
Still "used" here libraries\cms\html\sortablelist.php

**Comment by wilsonge** [comment]
Thankyou. Dirty bodge inbound. @dgrammatiko does this look like it would work?

**Comment by brianteeman** [comment]
can you fix the issue from drone please

File: libraries/cms/html/sortablelist.php [hunk 1]
```diff
@@ -43,42 +43,11 @@ abstract class JHtmlSortablelist
...
-				'button'     => $proceedSaveOrderButton
-			)
+		Factory::getDocument()->addScriptDeclaration('
+			const element = document.querySelector("' . $tableId . ' tbody");
+			element.classList.add("js-draggable");'
 		);
 
-		// Set static array
-		static::$loaded[__METHOD__] = true;
+		HtmlHelper::_('dragablelist.dragable', $tableId, $formId, $sortDir, $saveOrderingUrl, $proceedSaveOrderButton, $nestedList);
```
Comment (dgrammatiko) [inline libraries/cms/html/sortablelist.php:39]: I think only this line should be enough (iirc I had already code in the draggable script to cover this)

Comment (wilsonge) [reply libraries/cms/html/sortablelist.php:39]: I was using this https://github.com/joomla/joomla-cms/commit/17b8f67f54fd027ea15965ca8a492f4b3f55c971 as my basis which included adding the class

Comment (dgrammatiko) [reply libraries/cms/html/sortablelist.php:39]: This part https://github.com/joomla/joomla-cms/blob/b475126e0dbfd145e841c59978266cf5165341c9/build/media_source/system/js/draggable.es6.js#L13-L39 acts as a b/c for sortable. So even if the form doesn't have the class `js-draggable` will get it. for the script options that is also generated by the HTMLHelper for draggable. So, I think just aliasing this to `draggablelist.draggable` is fine

Comment (dgrammatiko) [reply libraries/cms/html/sortablelist.php:39]: Finally this https://github.com/joomla/joomla-cms/blob/b475126e0dbfd145e841c59978266cf5165341c9/libraries/cms/html/draggablelist.php#L58-L71 part wa supposed to act as the B/C part for the js

Comment (wilsonge) [reply libraries/cms/html/sortablelist.php:39]: I stand corrected :)

Comment (wilsonge) [reply libraries/cms/html/sortablelist.php:39]: You did even better than I thought before!!

File: libraries/cms/html/sortablelist.php [hunk 2]
```diff
@@ -43,42 +43,6 @@ abstract class JHtmlSortablelist
...
-		HTMLHelper::_('behavior.core');
-
-		// Depends on jQuery UI
-		HTMLHelper::_('jquery.ui', array('core', 'sortable'));
-
-		HTMLHelper::_('script', 'legacy/sortablelist.min.js', ['version' => 'auto', 'relative' => true]);
-		HTMLHelper::_('stylesheet', 'legacy/sortablelist.css', ['version' => 'auto', 'relative' => true]);
-
-		// Attach sortable to document
-		Factory::getDocument()->addScriptOptions(
```
Comment (Quy) [inline libraries/cms/html/sortablelist.php:68]: Remove `use Joomla\CMS\Factory;`.

Comment (Quy) [reply libraries/cms/html/sortablelist.php:68]: Delete lines 22-26?

Comment (wilsonge) [reply libraries/cms/html/sortablelist.php:68]: Done both

**Comment by wilsonge** [comment]
@zero-24 Can you do the deprecation merge in J3.10 please

**Comment by zero-24** [comment]
Done thanks

**Pull request merged**